src/rootcommands.h

```diff
@@ -6,12 +6,13 @@
 
 #include <functional>
 #include <memory>
+#include <string>
 #include <vector>
 
+#include "attribute.h"
 #include "types.h"
 
 class Root;
-class Attribute;
```

Comment:
Here, I also don't see why `class Attribute` should be replaced by `#include "attribute.h"`.